packages/jsdoc-plugins/lib/relation-fixer/getmissingdocletsdata.js

```diff
@@ -85,7 +85,7 @@ function getDocletsToAdd( allDoclets, childDoclet, options = {} ) {
 				}
 			}
 
-			return d.memberof === longname && isMatchingFilterOptions && !isUnwanted;
+			return isMatchingFilterOptions && !isUnwanted;
 		} );
 
 		docletsToAdd.push( ...toAdd );
```

Comment:
You're mixing functional programming with procedural programming. I'd either use `Array.filter()` or push elements to the original array.

E.g. return `docletsToAdd.concat( toAdd )` should be better.

Reply:
I see that this code wasn't touched in this PR. So, unless this causes some issues, it's beyond the scope of it.